packages/parser/tests/unit/schema/s3.test.ts

```diff
@@ -102,4 +104,21 @@ describe('S3 ', () => {
       parsed.userIdentity?.sessionContext?.attributes.mfaAuthenticated
     ).toEqual(false);
   });
+
+  it('should throw error when s3 event delete object does not contain etag or size', () => {
```

Comment:
How come did we add a new test? Did the formatting change the logic?

Reply:
I just saw your message on Discord! - let me take a look

Reply:
Ok, so, for everyone reading this later: it seems that we might have an inconsistency in the `S3RecordSchema` schema (to be confirmed).

We know from previous discussions ([here](https://github.com/aws-powertools/powertools-lambda-python/issues/4156#issuecomment-2097901803) & [here](https://github.com/aws-powertools/powertools-lambda-typescript/issues/2410#issue-2256226879)) that:
- `etag` (EventBridge->S3) and `eTag` (S3->Lambda/SNS/SQS) are omitted for Object Deletion events
- `size` is omitted for Object Deletion events

The `.refine()` method in the schema however seems to be asserting the opposite.

Looping in @am29d who knows this area of the codebase better than me.

Reply:
As discussed, we agreed to remove the `.refine()` method from the schema.